RecoPixelVertexing/PixelVertexFinding/test/VertexFinder_t.h

```diff
@@ -295,14 +375,16 @@ int main() {
         continue;
       }
 
-#ifdef __CUDACC__
+#ifndef CUDA_KERNELS_ON_CPU
       cudaCheck(cudaMemcpy(zv, LOC_ONGPU(zv), nv * sizeof(float), cudaMemcpyDeviceToHost));
       cudaCheck(cudaMemcpy(wv, LOC_ONGPU(wv), nv * sizeof(float), cudaMemcpyDeviceToHost));
```

Comment:
this is a typical case one does not want to make any memcpy on cpu....

Reply:
In general, the semantic can be one where making the copy is necessary also on cpu (e.g. because the function that launched the kernel does not keep it alive, or because the kernel makes changes to it that should not be reflected in the original buffer) or one where the copy is only required because of the different memory areas (e.g. the original buffer is guaranteed to stay alive, and the kernel does not make any changes to it).

Do you think it would make sense to define a couple of functions like `cudautils::copy` and `cudautils::mirror` ?
Then, the first could always be a copy (either `cudaMemcpy` or a plain copy) while the latter could be elided when running on the cpu.

Reply:
This is indeed a matter of discussion and prototyping.
I was thinking to some "magic" specialization of copy, and indeed your proposal is interesting as semantically expressive.

In reality in CMSSW production this never happens has
1) we copy to host memory using specific constructs by Matti
2) we do it explicitly in ad-hoc modules (SoAFromCUDA) and coded into the data format itself
2a) what I currently do is that the producer in the cpu WF have actually the name of the SoAFromCUDA producer/converter in the gpu WF so that the SoAonCPU have the same name in  both wf